lib/liquid/template.rb

```diff
@@ -78,17 +76,16 @@ def taint_mode=(mode)
         @taint_mode = mode
       end
 
+      # Sets the file system.
+      attr_writer :file_system
+
       attr_accessor :default_exception_renderer
       Template.default_exception_renderer = lambda do |exception|
         exception
       end
 
       def file_system
-        @@file_system
-      end
-
-      def file_system=(obj)
-        @@file_system = obj
+        @file_system ||= BlankFileSystem.new
```

Comment:
It would be simpler to just use `attr_accessor` and initialize the value at runtime.  That should avoid unnecessary runtime overhead as well.  Although this does follow the pattern of the existing code, so we could just change this along with the others in that way in a follow-up PR.

Reply:
It looks like `Template.default_exception_renderer` is already following the pattern I am suggesting.

Reply:
I see you opened https://github.com/Shopify/liquid/pull/1223

I'm good with that pattern too. As long that the whole class behave the same way for all attributes I'm good.